### PR TITLE
Clarified last_affected

### PR DESCRIPTION
**Pull request by hayleycd**
This is a first pass at further clarifying the `"last_affected"` field and addressing #146 and #150. 

Preview is available [here](https://hayleycd.github.io/osv-schema/#requirements)

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -650,7 +650,10 @@ Only **a single type** (either `"introduced"`, `"fixed"`, `"last_affected"`,
 Entries in the `events` array can contain either `"last_affected"` or `"fixed"`
 events, but not both. It's **strongly recommended** to use `"fixed"` instead of
 `"last_affected"` where possible, as it precisely identifies the version which
-contains the fix.
+contains the fix. `"last_affected"` should be thought of as a hard ceiling. 
+Versions above `"last_affected"` are considered unaffected. Unfortunately this 
+opens up the possibility for false negatives, which is why `"fixed"` is 
+overwhelmingly preferred.
```
Comment (chrisbloom7) [inline docs/schema.md:656]: I think it's useful to emphasize that the use of `last_affected` is limited to specific situations where there is no known fix at the time of publication. I also would suggest slightly de-emphasizing that versions above are unaffected, as likely as that scenario is.

```suggestion
contains the fix. `"last_affected"` should be thought of as the hard ceiling 
of the vulnerability _at the time of publication_ in the absence of a fixed version.
Versions above `"last_affected"` should be considered unaffected. Unfortunately
this opens up the possibility for false negatives, which is why `"fixed"` is 
overwhelmingly preferred.
```

**Review by chrisbloom7 — APPROVED** [review]
Thanks for adding this clarification! I added a small suggestion based on my [response](https://github.com/ossf/osv-schema/issues/150#issuecomment-1601930082) to the originating issue

File: docs/schema.md (hunk 1 above)
Comment (hayleycd) [reply docs/schema.md]: Agreed. Thanks!

**Review by oliverchang — APPROVED** [review]
Thanks @hayleycd and @chrisbloom7 for the wording fixes here!

**Pull request merged**